### ISSUE TITLE
Unable to `ALTER` JSON field with default value

### ISSUE THREAD

**Issue by bilogic** [doctrine/dbal#6037]
### Bug Report

|    Q        |   A
|------------ | ------
| Version     | 3.6.2

#### Current behaviour

```PHP
Schema::table('users', function (Blueprint $table) {
    $table->json('settings')->default(new Expression('(JSON_ARRAY())')); // runs perfect, no exceptions
});

Schema::table('users', function (Blueprint $table) {
    $table->json('settings')->nullable()->change(); // this line throws an exception
});
```

1. The above is code from a Laravel migration, which relies on this project.
2. It is giving me this exception `SQLSTATE[42000]: Syntax error or access violation: 1101 BLOB, TEXT, GEOMETRY or JSON column 'settings' can't have a default value (SQL: ALTER TABLE users CHANGE settings settings JSON DEFAULT 'json_array()')`
3. The problem is that the MySQL command is missing an extra set of brackets

I traced the culprit into DBAL. Adding the below code at the end of `getDefaultValueDeclarationSQL()` will solve the issue for MySQL, but I'm unsure if it will affect other databases.

```PHP
// add these 3 lines
if ($type instanceof Types\JsonType && in_array(strtoupper($default), ['JSON_ARRAY()', 'JSON_OBJECT()'])) {
    return ' DEFAULT ('.$default.')';
}

// before the final return
return ' DEFAULT '.$this->quoteStringLiteral($default);
```

https://github.com/doctrine/dbal/blob/4d27b9844749d611a963ea0d3caa4a9aec3fc1bb/src/Platforms/AbstractPlatform.php#L3108-L3112

#### Expected behaviour

`getDefaultValueDeclarationSQL()` should return ` DEFAULT (json_array())` instead of ` DEFAULT 'json_array()'`

In other words, `ALTER TABLE users CHANGE settings settings JSON DEFAULT (json_array())` is the correct MySQL command, whereas `ALTER TABLE users CHANGE settings settings JSON DEFAULT 'json_array()'` will throw the exception I reported

**Comment by bilogic** [comment]
related to #5689

**Comment by derrabus** [comment]
Closing as duplicate of #5689
